Yoast/wordpress-seo
# Remove target="_blank" from JS components and use makeOutboundLink instead

## Summary

This PR can be summarized in the following changelog entry:

- Improve links that open in a new tab

## Relevant technical choices:
- removes hardcoded `target="_blank"` attributes
- uses `makeOutboundLink` instead so there's now a proper visually hidden text to inform users
- reorders the imports for better clarity
- reduces the amount of ESLint warnings
- changes `disableKeyboardFocus` and `disableFocusRipple` values to make the `RaisedDefaultButton` focusable with a keyboard and have a focus style

The links in the Configuration Wizard step 10 are a special case, e.g.:

<img width="296" alt="Screenshot 2019-03-13 at 14 21 54" src="https://user-images.githubusercontent.com/1682452/54283733-ea537c80-459e-11e9-9de4-f94bfebf87a0.png">

Currently these links are actually button elements wrapped within links 😱 By changing the `url` prop to `href`, the material-ui component `RaisedButton` automatically renders a link instead of a button. No visual changes. This allows to remove the link we were wrapping around the button. `makeOutboundLink` does the rest, adding the appropriate attributes and the hidden a11y text.

## Test instructions
- `yarn && grunt build:js && grunt webpack:buildProd`
- test the links and check they have `target="_blank"` and the visually hidden text
  - in the configuration wizard error message: you can alter this line of code to make the error and link appear https://github.com/Yoast/wordpress-seo/blob/e0662b8e2bc141e474ec6db632a248f8d198c383/js/src/configuration-wizard.js#L124
  - in the dashboard widget: the Ryte "Analyze entire site" button-link
  - the button-links in the Configuration Wizard step 10

## UI changes

See #4605
